src/adagrams.js

```diff
@@ -1,15 +1,192 @@
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+const SCORE_POOL = {
+  A: 1,
+  B: 3,
+  C: 3,
+  D: 2,
+  E: 1,
+  F: 4,
+  G: 2,
+  H: 4,
+  I: 1,
+  J: 8,
+  K: 5,
+  L: 1,
+  M: 3,
+  N: 1,
+  O: 1,
+  P: 3,
+  Q: 10,
+  R: 1,
+  S: 1,
+  T: 1,
+  U: 1,
+  V: 4,
+  W: 4,
+  X: 8,
+  Y: 4,
+  Z: 10,
+};
+//############ WAVE 1 ###############
+
+/**
+ * Function takes object.
+ * Returns random key from dictionary.
+ */
+const getRandomLetter = (letterPool) => {
+  const keys = Object.keys(letterPool);
+  return keys[Math.floor(Math.random() * keys.length)];
+};
+
+/**
+ *Return an array of ten strings from the LETTER_POOL.
+ */
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const copyLetterPool = { ...LETTER_POOL };
+  const arrayLetters = [];
+  while (arrayLetters.length < 10) {
+    let letter = getRandomLetter(copyLetterPool);
+    arrayLetters.push(letter);
+    copyLetterPool[letter] -= 1;
+    if (copyLetterPool[letter] === 0) {
+      delete copyLetterPool[letter];
+    }
+  }
+  return arrayLetters;
 };
 
+//############ WAVE 2 ###############
+// Solution # 1
+/**
+ * Returns true if every letter in the input word is available
+ * (in the right quantities) in the lettersInHand.
+ */
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  for (const i in input) {
+    let letter = input[i].toUpperCase();
+    if (!lettersInHand.includes(letter)) return false;
+    else delete lettersInHand[i];
```

Comment:
This function works well, but it has the side effect of changing `lettersInHand`. I recommend avoiding side effects unless they are an intended part of the code.